plugins/scale/scale.cpp

```diff
@@ -878,26 +874,34 @@ class wayfire_scale : public wf::plugin_interface_t
 
         filter_views(views);
 
+        wf::scale_transformer::padding pad;
+        scale_padding_signal signal(pad);
+        output->emit_signal("scale-padding", &signal);
+
         auto workarea = output->workspace->get_workarea();
 
         auto sorted_rows = view_sort(views);
         size_t cnt_rows  = sorted_rows.size();
 
-        const double scaled_height = (double)
-            (workarea.height - (cnt_rows + 1) * spacing) / cnt_rows;
+        const double scaled_height = std::max((double)
+            (workarea.height - (cnt_rows + 1) * spacing) / cnt_rows -
+            pad.top - pad.bottom, 1.0);
         current_row_sizes.clear();
 
         for (size_t i = 0; i < cnt_rows; i++)
         {
             size_t cnt_cols = sorted_rows[i].size();
             current_row_sizes.push_back(cnt_cols);
-            const double scaled_width = (double)
-                (workarea.width - (cnt_cols + 1) * spacing) / cnt_cols;
+            const double scaled_width = std::max((double)
+                (workarea.width - (cnt_cols + 1) * spacing) / cnt_cols -
+                pad.left - pad.right, 1.0);
 
             for (size_t j = 0; j < cnt_cols; j++)
             {
-                double x = workarea.x + spacing + (spacing + scaled_width) * j;
-                double y = workarea.y + spacing + (spacing + scaled_height) * i;
+                double x = workarea.x + spacing +
```

Comment:
Maybe I am misunderstanding something, but shouldn't `pad_left/right` be added only once (not multiplied by `j`)?

Reply:
My assumption here is that each view has the same padding around it. This makes padding less flexible though.

Reply:
Wouldn't it be better if we could do this: for each view, have a fixed size available for view + padding (for example, width for each view is `workarea_width / views_in_the_row`), and then, the width of the view is the `max_width - spacing - padding`?